File: src/Steak.cpp
#include "../include/Steak.hpp"

LinkedList<FarmProductType> Steak::resep;

//Konstruktor Steak
Steak::Steak(int _price)  {
    initializeProduct(_price, SIDEPRODUCT, STEAK, "Steak");
    InitResep();
}

void Steak::InitResep() {
    //Inisialisasi resep:
    resep.add(MEAT);
    resep.add(MILK);
}

FarmProductType Steak::getResepwithIdx(int i) {
    return Steak::getResep().get(i);
}

File: Java/src/main/java/engisubs/gameobject/product/side_product/Steak.java
package main.java.engisubs.gameobject.product.side_product;

import java.util.LinkedList;

public class Steak extends SideProduct{
    private static LinkedList<FarmProductType> resep;

    public Steak(int _price){
        super.initializeProduct(_price, ProductType.SIDEPRODUCT, GameObjectType.STEAK, "Medium Rare Espanolia Steak~~");
    }

    public static LinkedList<FarmProductType> getResep(){
        return resep;
    }

    public static FarmProductType getResepwithIdx(int index){
        return getResep().get(index);
    }

    public static void InitResep(){
        resep.add(FarmProductType.MEAT);
        resep.add(FarmProductType.EGG);
    }
}